src/transforms/levels.jl

```diff
@@ -0,0 +1,72 @@
+"""
+    Levels(:a => ["yes,"no"])
+
+Return a copy of the table with specified levels and orders for categorical columns
+allowing only changing the order of the column
+
+# Examples
+
+```julia
+Levels(:a => ["yes,"no"], :c => [1,2,4], :d = ["a","b","c"])
+Levels("a" => ["yes","no"], "c" => [1,2,4], ordered= ["a","c"])
+Levels(:a => ["yes","no"], :c => [1,2,4], :d => [1,23,5,7] ,ordered = [:a,:b,:c])
+```
+"""
+
+@kwdef struct Levels{K} <: Stateless
+  levelspec::K
+  ordered::AbstractVector{Symbol}
+end
+#should work more on constructors
+Levels(pairs::Pair{K}...; ordered::AbstractVector{K}=Symbol[]) where {K <: Symbol} =
+  Levels(NamedTuple(pairs), ordered)
+
+Levels(pairs::Pair{K}...; ordered::AbstractVector{K}=String[]) where {K <: AbstractString} =
+  Levels(NamedTuple(Symbol(k) => v for (k,v) in pairs), Symbol.(ordered))
+
+
+
+isrevertible(transform::Levels) = true
+# handle three cases
+#1. when the col is already a categorical array and wanna change levels and order
+categorify(level::AbstractVector, x::CategoricalVector, ordered::Bool) = levels(x), categorical(x, levels = level, ordered=ordered) 
+#2. when the col is normal array and want to change to categorical array
+categorify(level::AbstractVector, x::AbstractVector, ordered::Bool) = unwrap, categorical(x,levels = level, ordered=ordered) 
+#3. when the col is not need for change or convert back to normal array
+categorify(func::Function, x::AbstractVector, ordered::Bool) = ordered ? (levels(x), categorical(x, ordered=true)) : (func, func.(x))
```

Comment:
```suggestion
# when the col is already a categorical array and wanna change levels and order
categorify(l, x::CategoricalVector, o) = levels(x), categorical(x, levels=l, ordered=o)

# when the col is normal array and want to change to categorical array
categorify(l, x::AbstractVector, o) = unwrap, categorical(x, levels=l, ordered=o)

# when the col is not need for change or convert back to normal array
categorify(f::Function, x::AbstractVector, o) = o ? (levels(x), categorical(x, ordered=true)) : (f, f.(x))
```

Reply:
It seems that we have to specify the level type for the first one to be `AbstractVector` as when `identity` or `unwrap` function get called on `CategoricalArray` it would be ambiguous

Reply:
Thanks for checking. Please add the minimum number of type annotations that can disambiguate the methods.